docs/architecture/adr-066-keyring-plugin.md

```diff
@@ -0,0 +1,451 @@
+# ADR-066: Keyring plugin
+
+## Changelog
+
+* Jun 20, 2023: Initial Draft (@JulianToledano & @bizk)
+
+## Status
+
+DRAFT
+
+## Abstract
+
+This ADR describes a keyring implementation based on the hashicorp plugins over gRPC.
+
+## Context
+
+Currently, in the cosmos-sdk, the keyring implementation depends on
+[99designs/keyring](https://github.com/99designs/keyring) module that isn't under actively maintenance.
+
+For that reason, it is proposed to develop a new keyring implementation that leverages
+HashiCorp plugins over gRPC. These can be abstracted and implemented in any language,
+while maintaining its capacity to use the rest of the cosmos-sdk features.
+This approach adds extendability to the system itself, and provides more adoption.
+
+## Alternatives
+
+There are several other options available, such as plugins over RPC or utilizing the Go standard
+library plugin package. However, in the end, all these plugin alternatives share similarities.
+
+Another alternative is to reimplement the keystore [db](https://github.com/cosmos/cosmos-sdk/blob/v0.47.3/crypto/keyring/keyring.go#L207)
+interface. This db is where the `99designs` dependency lays. 
+
+## Decision
+
+We will define the gRPC service with all the necessary messages. Additionally, we will
+implement one plugin for each backend currently supported in the current keyring.
+
+
+### gRPC Definitions
+
+The idea is to define a service method for each method already present in the keyring interface.
+Similarly, a request/response message will be defined for each service method.
+
+#### Service Definition
+```protobuf
+service KeyringService {
+  rpc Backend(BackendRequest) returns (BackendResponse);
```

Comment:
why does backend need to be part of this api?

Reply:
This method is meant to return the plugins' name and version.

Reply:
why does the node need to know what backend is being used?